### ISSUE TITLE
All solid edges shown as occluded

### ISSUE THREAD

**Issue by Symbian9** [solvespace/solvespace#378]
### System information

SolveSpace version: 3.0~10123738

Operating system: Debian 9.x Stretch

### Expected behavior

Only really occluded edges should be shown as hidden

### Actual behavior

All solid edges shown as occluded

### Additional information

![pic.1](https://i.imgur.com/zlxb8iy.png)

![pic.2](https://i.imgur.com/J9hA9xw.png)

![pic.3](https://i.imgur.com/Zbx17WO.png)

**Comment by Evil-Spirit** [comment]
@whitequark probably here is the problem caused by https://github.com/solvespace/solvespace/commit/e383b7fba8fbad23ac370ea5731a87b7eacdf6c8

**Comment by whitequark** [comment]
@Evil-Spirit No, it was a problem as far as I can remember, I just never paid attention to it. It renders the same way if I revert e383b7f.

**Comment by phkahler** [comment]
On my system it looks like the occluded lines are drawn by drawing all lines (in whichever style) without Z-buffer, and then drawing all of them again in normal style with the Z-buffer enabled. It looks like some kind of fighting between the two line styles rather than only using the right one for each edge. Is that what's happening? @Evil-Spirit ?

**Comment by Evil-Spirit** [comment]
Probably, I can blame glDepthRangef follwing after this. I suppose for hidden lines it should be performed in a different way.

**Comment by Evil-Spirit** [comment]
Also, depth write probably should be switched off  (glDepthMask(GL_FALSE)) for non-occluded edges.

**Comment by phkahler** [comment]
@Evil-Spirit its behaving like depth testing is turned off for occluded edges. And yes, depth write should probably be off for all edges.

**Comment by phkahler** [comment]
The problem has to be that there are multiple areas in the code affecting this. I can go in Group::Draw(Canvas *canvas) and comment out the entire section enclosed by  if(SS.GW.showEdges) {  ....  } and I'm still able to use the toggle buttons to produce most of the options including showing the occluded lines. The bug is because a lot of this section of code is redundant in some way - I just haven't figured out what's stepping on it.

**Comment by phkahler** [comment]
OK, commenting out everyting but:
```
    GenerateDisplayItems();
    DrawMesh(DrawMeshAs::DEFAULT, canvas);
```
It's still drawing almost all edges (no silhouettes) and allowing 3 options for them with zero calls to DrawOutlines. This is also enabled/disabled via the "show all / hide all" links in the text window. I thought those were only supposed to toggle entities and the toolbar buttons determined which edges are visible and what style.
